jackson/src/test/kotlin/dev/codified/jackson/CodifiedJacksonModuleTest.kt

```diff
@@ -0,0 +1,154 @@
+package dev.codified.jackson
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import org.junit.jupiter.api.Test
+import pl.brightinventions.codified.Codified
+import pl.brightinventions.codified.enums.CodifiedEnum
+import pl.miensol.shouldko.shouldEqual
+
+class CodifiedJacksonModuleTest {
+    val objectMapper = ObjectMapper().findAndRegisterModules()
```

Comment:
If we would like to add some tests for `data class` as well, we could add another `testImplementation` dependency: `"com.fasterxml.jackson.module:jackson-module-kotlin:${Versions.jackson}"` and then update this `objectMapper` here:

```suggestion
    val objectMapper = jacksonObjectMapper().findAndRegisterModules()
```

Then, this should work:

```kotlin
data class HasSizeAndColour(
    val size: Size?,
    val colour: Colour?,
)

data class HasCodifiedSizeAndColour(
    val size: CodifiedEnum<Size, String>,
    val colour: CodifiedEnum<Colour, String>,
)


@Test
fun `can serialize and deserialize two known values to enum properties`() {
    val input = HasSizeAndColour(Size.Large, Colour.Green)

    val output = serde(input)

    output.size.shouldEqual(Size.Large)
    output.colour.shouldEqual(Colour.Green)
}

@Test
fun `can serialize and deserialize two known values to codified enum properties`() {
    val input = HasCodifiedSizeAndColour(Size.Medium.codifiedEnum(), Colour.Red.codifiedEnum())

    val output = serde(input)

    output.size.knownOrNull().shouldEqual(Size.Medium)
    output.colour.knownOrNull().shouldEqual(Colour.Red)
}

@Test
fun `can serialize and deserialize two not known values to codified enum properties`() {
    val input = HasCodifiedSizeAndColour("XS".codifiedEnum(), "black".codifiedEnum())

    val output = serde(input)

    output.size.knownOrNull().shouldEqual(null)
    output.size.code().shouldEqual("XS")
    output.colour.knownOrNull().shouldEqual(null)
    output.colour.code().shouldEqual("black")
}

@Test
fun `can deserialize two not known values to codified enum properties`() {
    val input = """{"size": "XS", "colour": "black"}"""

    val output = objectMapper.readValue(input, HasCodifiedSizeAndColour::class.java)

    output.size.knownOrNull().shouldEqual(null)
    output.size.code().shouldEqual("XS")
    output.colour.knownOrNull().shouldEqual(null)
    output.colour.code().shouldEqual("black")
}

@Test
fun `can deserialize two not known values to enum properties`() {
    val input = """{"size": "XS", "colour": "black"}"""

    val output = objectMapper.readValue(input, HasSizeAndColour::class.java)

    output.size.shouldEqual(null)
    output.colour.shouldEqual(null)
}
```

Feel free to add them if you think it's useful.

Reply:
> If we would like to add some tests for `data class` as well, we could add another `testImplementation` dependency: `"com.fasterxml.jackson.module:jackson-module-kotlin:${Versions.jackson}"` and then update this `objectMapper` here:
> 
> Then, this should work:
> 
> ```kotlin
> data class HasSizeAndColour(
>     val size: Size?,
>     val colour: Colour?,
> )
> 
> data class HasCodifiedSizeAndColour(
>     val size: CodifiedEnum<Size, String>,
>     val colour: CodifiedEnum<Colour, String>,
> )
> 
> 
> @Test
> fun `can serialize and deserialize two known values to enum properties`() {
>     val input = HasSizeAndColour(Size.Large, Colour.Green)
> 
>     val output = serde(input)
> 
>     output.size.shouldEqual(Size.Large)
>     output.colour.shouldEqual(Colour.Green)
> }
> 
> @Test
> fun `can serialize and deserialize two known values to codified enum properties`() {
>     val input = HasCodifiedSizeAndColour(Size.Medium.codifiedEnum(), Colour.Red.codifiedEnum())
> 
>     val output = serde(input)
> 
>     output.size.knownOrNull().shouldEqual(Size.Medium)
>     output.colour.knownOrNull().shouldEqual(Colour.Red)
> }
> 
> @Test
> fun `can serialize and deserialize two not known values to codified enum properties`() {
>     val input = HasCodifiedSizeAndColour("XS".codifiedEnum(), "black".codifiedEnum())
> 
>     val output = serde(input)
> 
>     output.size.knownOrNull().shouldEqual(null)
>     output.size.code().shouldEqual("XS")
>     output.colour.knownOrNull().shouldEqual(null)
>     output.colour.code().shouldEqual("black")
> }
> 
> @Test
> fun `can deserialize two not known values to codified enum properties`() {
>     val input = """{"size": "XS", "colour": "black"}"""
> 
>     val output = objectMapper.readValue(input, HasCodifiedSizeAndColour::class.java)
> 
>     output.size.knownOrNull().shouldEqual(null)
>     output.size.code().shouldEqual("XS")
>     output.colour.knownOrNull().shouldEqual(null)
>     output.colour.code().shouldEqual("black")
> }
> 
> @Test
> fun `can deserialize two not known values to enum properties`() {
>     val input = """{"size": "XS", "colour": "black"}"""
> 
>     val output = objectMapper.readValue(input, HasSizeAndColour::class.java)
> 
>     output.size.shouldEqual(null)
>     output.colour.shouldEqual(null)
> }
> ```
> 
> Feel free to add them if you think it's useful.

Yes. I was considering it. However, then it hit me that our production code has no dependency on `jackson-module-kotlin`. I'm planning to use the library in Java. Hence first and foremost I'm targeting testing with as little dependency on Kotlin as possible.